R/report_dispatcher.R

```diff
@@ -13,15 +13,25 @@
 #' \code{studentids, depth_string} into the arg list, as well as named elements 
 #' corresponding to the key/value cut and element outlined above
 #' @param calling_env defaults to parent frame.
+#' @param post_process a post processing function to apply to the list of plots we get back.
+#' default behavior is only_valid_plots(), which drops any plot that failed.  
+#' don't want that?  write something new :)
+#' @param verbose should the function print updates about what is happening?  default is TRUE.
 #' @param ... other parameters to pass through (namely cdfs).  todo: reformat this function
 #' to take mapvizieR object.
 #' 
 #' @export
 #' 
-#' @return a cdf with new term fields 
+#' @return a list of output from the function you called
 
-report_dispatcher <- function(mapvizieR_obj, cut_list, call_list, 
-    func_to_call, arg_list=list(), calling_env = parent.frame(), ...
+report_dispatcher <- function(
+    mapvizieR_obj,
+    cut_list, call_list, 
+    func_to_call, arg_list=list(),
+    calling_env = parent.frame(),
+    post_process = "only_valid_plots",
```

Comment:
Not sure if it matters.  My gut instinct is that passing in unquoted
functions is best, since it points the user in the direction of write your
own function and makes it a bit (only a bit, though) more obvious that it
is a function, rather some string that is going to trip some conditional
statement somewhere internal to the report_dispatcher.

*Chris Haid*

Director of Research and Analysis

KIPP Chicago

312.961.0510

chaid@kippchicago.org

*Work hard*. *Be nice.*

On Fri, Jan 30, 2015 at 1:59 PM, Andrew Martin <notifications@github.com>
wrote:

> In R/report_dispatcher.R
> <https://github.com/almartin82/mapvizieR/pull/36#discussion_r23868085>:
>
> >
> > -report_dispatcher <- function(mapvizieR_obj, cut_list, call_list,
> > -    func_to_call, arg_list=list(), calling_env = parent.frame(), ...
> > +report_dispatcher <- function(
> > +    mapvizieR_obj,
> > +    cut_list, call_list,
> > +    func_to_call, arg_list=list(),
> > +    calling_env = parent.frame(),
> > +    post_process = "only_valid_plots",
>
> hmm, got it. does it matter either way? house style?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/almartin82/mapvizieR/pull/36/files#r23868085>.
>